datafusion/core/src/physical_planner.rs

```diff
@@ -785,26 +797,27 @@ impl DefaultPhysicalPlanner {
                     output_schema: Arc::new(schema),
                     table_partition_cols,
                     overwrite: false,
+                    keep_partition_by_columns,
                 };
                 let mut table_options = session_state.default_table_options();
                 let sink_format: Arc<dyn FileFormat> = match format_options {
                     FormatOptions::CSV(options) => {
                         table_options.csv = options.clone();
                         table_options.set_file_format(FileType::CSV);
-                        table_options.alter_with_string_hash_map(source_option_tuples)?;
+                        table_options.alter_with_string_hash_map(&updated_source_options_tuples)?;
```

Comment:
It would be nice to not have to allocate a new hashmap. Were you getting an error if the new option was left in the hashmap?

Reply:
I refactored this to have two hashmaps from the beginning, instead of re-copying values.